diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN hal-0.5.12.orig/hal.conf.in hal-0.5.12/hal.conf.in
--- hal-0.5.12.orig/hal.conf.in	2009-02-04 17:07:23.000000000 -0500
+++ hal-0.5.12/hal.conf.in	2009-05-23 12:54:16.000000000 -0400
@@ -14,8 +14,8 @@
     <allow own="org.freedesktop.Hal"/>
   </policy>
 
-  <!-- Allow anyone to invoke methods on the Manager and Device interfaces -->
-  <policy context="default">
+  <!-- Gentoo specific rules to allow 'plugdev' group to access Volumes and Power Management -->
+  <policy group="plugdev">
     <allow send_destination="org.freedesktop.Hal"
            send_interface="org.freedesktop.DBus.Introspectable"/>
     <allow send_destination="org.freedesktop.Hal"
@@ -52,6 +52,8 @@
     	   send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
     <allow send_destination="org.freedesktop.Hal"
     	   send_interface="org.freedesktop.Hal.Device.WakeOnLan"/>
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.Hal.Device.VideoAdapterPM"/>
 
   </policy>
 
