diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN hal-0.5.13.orig/hal.conf.in hal-0.5.13/hal.conf.in
--- hal-0.5.13.orig/hal.conf.in	2009-02-04 17:07:23.000000000 -0500
+++ hal-0.5.13/hal.conf.in	2009-07-22 10:35:19.985169862 -0400
@@ -10,12 +10,35 @@
   <policy user="@HAL_USER@">
     <allow own="org.freedesktop.Hal"/>
   </policy>
-  <policy user="root">
-    <allow own="org.freedesktop.Hal"/>
+
+  <!-- root needs access to things for X, automount, powermanagement, etc. -->
+  <policy user="root">
+    <allow own="org.freedesktop.Hal"/>
+    <allow send_destination="org.freedesktop.Hal"
+        send_interface="org.freedesktop.Hal.Device"/>
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.Hal.Manager"/>
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.Hal.Device.KillSwitch"/>
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.Hal.Device.Leds"/>
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.Hal.Device.Storage"/>
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.Hal.Device.Storage.Removable"/>
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.Hal.Device.Volume"/>
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.Hal.Device.WakeOnLan"/>
   </policy>
 
-  <!-- Allow anyone to invoke methods on the Manager and Device interfaces -->
-  <policy context="default">
+
+  <!-- Gentoo specific rules to allow 'plugdev' group to access Volumes and Power Management -->
+  <policy group="plugdev">
     <allow send_destination="org.freedesktop.Hal"
            send_interface="org.freedesktop.DBus.Introspectable"/>
     <allow send_destination="org.freedesktop.Hal"
@@ -52,6 +75,8 @@
     	   send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
     <allow send_destination="org.freedesktop.Hal"
     	   send_interface="org.freedesktop.Hal.Device.WakeOnLan"/>
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.Hal.Device.VideoAdapterPM"/>
 
   </policy>
 
