diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN hal-0.5.12rc1.orig/hal.conf.in hal-0.5.12rc1/hal.conf.in
--- hal-0.5.12rc1.orig/hal.conf.in	2008-11-27 09:32:24.000000000 -0500
+++ hal-0.5.12rc1/hal.conf.in	2009-05-23 12:18:45.000000000 -0400
@@ -15,36 +15,47 @@
   </policy>
 
   <!-- Allow anyone to invoke methods on the Manager and Device interfaces -->
+  <!-- Policykit regulates actual access -->
   <policy context="default">
-    <allow send_interface="org.freedesktop.Hal.Manager"/>
-    <allow send_interface="org.freedesktop.Hal.Device"/>
-    <allow receive_interface="org.freedesktop.Hal.Manager"
-           receive_sender="org.freedesktop.Hal"/>
-    <allow receive_interface="org.freedesktop.Hal.Device"
-           receive_sender="org.freedesktop.Hal"/>
-
-    <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
-    <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
-    <allow send_interface="org.freedesktop.Hal.Device.Volume"/>
-    <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
-    <allow receive_interface="org.freedesktop.Hal.Device.SystemPowerManagement"
-	   receive_sender="org.freedesktop.Hal"/>
-    <allow receive_interface="org.freedesktop.Hal.Device.LaptopPanel"
-	   receive_sender="org.freedesktop.Hal"/>
-    <allow receive_interface="org.freedesktop.Hal.Device.Volume"
-	   receive_sender="org.freedesktop.Hal"/>
-    <allow receive_interface="org.freedesktop.Hal.Device.Volume.Crypto"
-	   receive_sender="org.freedesktop.Hal"/>
-  </policy>
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.DBus.Introspectable"/>
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.DBus.Properties" />
+
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.Hal.Device"/>
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.Hal.Manager"/>
+
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.Hal.Device.CPUFreq"/>
+    <allow send_destination="org.freedesktop.Hal"
+    	   send_interface="org.freedesktop.Hal.Device.DockStation"/>
+    <allow send_destination="org.freedesktop.Hal"
+    	   send_interface="org.freedesktop.Hal.Device.KillSwitch"/>
+    <allow send_destination="org.freedesktop.Hal"
+    	   send_interface="org.freedesktop.Hal.Device.KeyboardBacklight"/>
+    <allow send_destination="org.freedesktop.Hal"
+    	   send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
+    <allow send_destination="org.freedesktop.Hal"
+    	   send_interface="org.freedesktop.Hal.Device.Leds"/>
+    <allow send_destination="org.freedesktop.Hal"
+    	   send_interface="org.freedesktop.Hal.Device.LightSensor"/>
+    <allow send_destination="org.freedesktop.Hal"
+    	   send_interface="org.freedesktop.Hal.Device.Storage"/>
+    <allow send_destination="org.freedesktop.Hal"
+    	   send_interface="org.freedesktop.Hal.Device.Storage.Removable"/>
+    <allow send_destination="org.freedesktop.Hal"
+    	   send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
+    <allow send_destination="org.freedesktop.Hal"
+    	   send_interface="org.freedesktop.Hal.Device.Volume"/>
+    <allow send_destination="org.freedesktop.Hal"
+    	   send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
+    <allow send_destination="org.freedesktop.Hal"
+    	   send_interface="org.freedesktop.Hal.Device.WakeOnLan"/>
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.Hal.Device.VideoAdapterPM"/>
 
-  <!-- Default policy for the exported interfaces; if PolicyKit is not used
-       for access control you will need to modify this -->
-  <policy context="default">
-    <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
-    <allow send_interface="org.freedesktop.Hal.Device.VideoAdapterPM"/>
-    <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
-    <allow send_interface="org.freedesktop.Hal.Device.Volume"/>
-    <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
   </policy>
 
 </busconfig>
