diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN hal-0.5.12.orig/hal.conf.in hal-0.5.12/hal.conf.in
--- hal-0.5.12.orig/hal.conf.in	2009-02-04 17:07:23.000000000 -0500
+++ hal-0.5.12/hal.conf.in	2009-05-23 12:58:33.000000000 -0400
@@ -15,6 +15,7 @@
   </policy>
 
   <!-- Allow anyone to invoke methods on the Manager and Device interfaces -->
+  <!-- Policykit regulates actual access -->
   <policy context="default">
     <allow send_destination="org.freedesktop.Hal"
            send_interface="org.freedesktop.DBus.Introspectable"/>
@@ -52,6 +53,8 @@
     	   send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
     <allow send_destination="org.freedesktop.Hal"
     	   send_interface="org.freedesktop.Hal.Device.WakeOnLan"/>
+    <allow send_destination="org.freedesktop.Hal"
+           send_interface="org.freedesktop.Hal.Device.VideoAdapterPM"/>
 
   </policy>
 
