diff -uNr hal-0.5.11.ORIG/hald/hald-cache-test.sh hal-0.5.11/hald/hald-cache-test.sh
--- hal-0.5.12rc1.ORIG/hald/hald-cache-test.sh	2009-02-04 22:26:18.000000000 +0000
+++ hal-0.5.12rc1/hald/hald-cache-test.sh	2009-02-04 22:27:13.000000000 +0000
@@ -3,9 +3,9 @@
 mkdir .local-fdi-test
 make -C ../fdi install DESTDIR=`pwd`/.local-fdi-test prefix=/
 
-HAL_FDI_SOURCE_PREPROBE=.local-fdi-test/share/hal/fdi/preprobe
-HAL_FDI_SOURCE_INFORMATION=.local-fdi-test/share/hal/fdi/preprobe
-HAL_FDI_SOURCE_POLICY=.local-fdi-test/share/hal/fdi/policy
+HAL_FDI_SOURCE_PREPROBE=.local-fdi-test/usr/share/hal/fdi/preprobe
+HAL_FDI_SOURCE_INFORMATION=.local-fdi-test/usr/share/hal/fdi/preprobe
+HAL_FDI_SOURCE_POLICY=.local-fdi-test/usr/share/hal/fdi/policy
 HAL_FDI_CACHE_NAME=.local-fdi-test/hald-local-fdi-cache
 export HAL_FDI_SOURCE_PREPROBE HAL_FDI_SOURCE_INFORMATION \
        HAL_FDI_SOURCE_POLICY HAL_FDI_CACHE_NAME
